### ISSUE TITLE
L7: Softmax Quiz lecture text, row or column

### ISSUE THREAD

**Issue by HaiShaw** [udacity/sdc-issue-reports#91]
I'm not sure if the text row of below is correct, may it be column?

Given a two-dimensional array where each row represents a set of logits:

# logits is a two-dimensional array
logits = np.array([
    [1, 2, 3, 6],
    [2, 4, 5, 6],
    [3, 8, 7, 6]])
# softmax will return a two-dimensional array with the same shape
print softmax(logits)

[
    [ 0.09003057  0.00242826  0.01587624  0.33333333]
    [ 0.24472847  0.01794253  0.11731043  0.33333333]
    [ 0.66524096  0.97962921  0.86681333  0.33333333]
]

**Comment by Brok-Bucholtz** [comment]
@HaiShaw 

I'm sorry, I'm having a hard time understanding the issue.  Can you tell me what you think the array should be?

**Comment by HaiShaw** [comment]
If each row represent one set of logits, then calculated probability is not wrong, the implemented code is thinking each column is one logits.

**Comment by HaiShaw** [comment]
If each row represent one set of logits, then calculated probability is wrong, the implemented code in place is thinking each column is one logits.
This has to do np matrix operations.

**Comment by HaiShaw** [comment]
Got to unify the input to array, keep it flexible either list or an array fine - only if in the function we can convert to array uniformly correct, in other words, if input is list
input = [1,2,3]

need to convert it to [[1,2,3]] in shape of (1,3), where row is a sample.

then it is consistent with 

logits = np.array([
[1, 2, 3, 6],
[2, 4, 5, 6],
[3, 8, 7, 6]])

where each row is a sample - logit as lecture says.

**Comment by HaiShaw** [comment]
Look at the next quiz (L7.15), tf.nn.softmax() is always correct regardless input list of 2d array:

{code}
# Solution is available in the other "solution.py" tab
import tensorflow as tf
import numpy as np


def run():
    output = None
    
    #logit_data = [2.0, 1.0, 0.1]
    
    logit_data = np.array([
        [1, 2, 3, 6],
        [2, 4, 5, 6],
        [3, 8, 7, 6]])
        
    logits = tf.placeholder(tf.float32)
    
    # ToDo: Calculate the softmax of the logits
    softmax = tf.nn.softmax(logits)     
    
    with tf.Session() as sess:
        # ToDo: Feed in the logits data
        output = sess.run(softmax, feed_dict={logits: logit_data} )

    return output

{code}

it gives:
[[ 0.00626879  0.01704033  0.04632042  0.93037045]
 [ 0.01203764  0.08894681  0.24178252  0.657233  ]
 [ 0.00446236  0.66227239  0.2436364   0.08962882]]

**Comment by Brok-Bucholtz** [comment]
@HaiShaw 

Thank you for the explanation, it was really helpful!  I'm going to look into this.

**Comment by Brok-Bucholtz** [comment]
@HaiShaw 

I fixed the text to say each column in the 2d array is a set of logits, instead of the row.  Let me know if there is anything else and thanks for your help!

**Comment by HaiShaw** [comment]
@Brok-Bucholtz 

My pleasure. Thanks for your prompt follow up!

**Comment by Brok-Bucholtz** [comment]
@HaiShaw 

It's my pleasure as well.  Thank you for taking your time to write to us!
